test/pathforaSpec.js

```diff
@@ -1162,17 +1162,57 @@ describe('API', function () {
     var widget = $('#' + form.id);
     expect(widget.length).toBe(0);
   });
-  
-  if('should not show an impression counter widget without an id', function () {
+
+  it('should not show a widget with hideAfterAction without an id', function () {
+    expect(function() {
+      var form = new window.pathfora.Form({
+        msg: 'subscription',
+        header: 'Header',
+        layout: 'slideout',
+        position: 'bottom-right',
+        displayConditions: {
+          hideAfterAction: {
+            confirm: true
+          }
+        }
+      });
+    }).toThrow(new Error('Widgets with the hideAfterAction displayConditions need a preset id value. Display condition denied.'));
+  });
+
+  it('should show widget with hideAfterAction after no action', function () {
+    var widgetId = 'hideAfterActionWidget1';
     var form = new window.pathfora.Form({
+      id: widgetId,
       msg: 'subscription',
       header: 'Header',
       layout: 'slideout',
       position: 'bottom-right',
       displayConditions: {
-        impressions: {
-          session: 1,
-          total: 5
+        hideAfterAction: {
```

Comment:
Yeah, I had a similar note in the pr description, I think limiting the number of times it could show after each action too. So combining these into maybe something like:

```
hideAfterAction: {
  closed: {
    hideCount: 3,
    duration: 604800 // one week
  },
  confirm: {
    hideCount: 1
  }
}
``` 
In this case the modal would show again in a week after the first close, another week after the second, and then hide permanently (the length of the cookie) after the third close, however if the confirm button is ever clicked the modal will hide permanently (the length of the cookie).

Though now that I think about it I wonder if maybe this is too much, do you think people will care about the number of times they click "x"? Because the hideAfterAction duration would also work nicely with impressions. I'll implement what you suggested for now, but I'd like to get your thoughts on the above as well.